test/e2e/service.go

```diff
@@ -2764,3 +2821,23 @@ func describeSvc(ns string) {
 		"describe", "svc", fmt.Sprintf("--namespace=%v", ns))
 	framework.Logf(desc)
 }
+
+func checkReachability(expectToBeReachable bool, namespace, pod, target string) {
+	cmd := fmt.Sprintf("wget -T 5 -qO- %q", target)
+	if expectToBeReachable {
+		err := wait.PollImmediate(framework.Poll, 2*time.Minute, func() (bool, error) {
+			_, err := framework.RunHostCmd(namespace, pod, cmd)
+			if err != nil {
+				framework.Logf("got err: %v, retry until timeout", err)
+				return false, err
```

Comment:
We should `return false, nil` here, otherwise this poll loop will exit at the first failure.